src/client.rs

```diff
@@ -1146,6 +1171,43 @@ impl QdrantClient {
 
         Ok(())
     }
+
+    pub async fn upload_snapshot<T>(
+        &self,
+        in_path: impl Into<PathBuf>,
+        collection_name: T,
+        rest_api_uri: Option<T>,
+    ) -> Result<()>
+    where
+        T: ToString + Clone,
+    {
+        let snapshot_path: PathBuf = in_path.into();
+        let filename = snapshot_path
+            .file_name()
+            .and_then(|name| name.to_str().map(|name| name.to_string()));
+
+        let snapshot_file = std::fs::read(snapshot_path)?;
```

Comment:
nitpick: we could read the file using tokio::fs to run on a different thread pool

Reply:
I saw the `std::fs` being used in the `download_snapshot()` and thought this is the preferred way of writing files.

Reply:
You can leave like it this :+1: 
We will migrate all call sites throughout the code to the async. version if it becomes a problem later :+1:

Reply:
It's alright. I changed it in both the upload and download. Since there's no other file operations in the rust client, I thought it'd be a good idea to eliminate all `std::fs` from the beginning. Especially since `tokio::fs` is a drop-in replacement so testing that it still works as expected was simple enough.